tokio-trace-fmt/src/filter.rs

```diff
@@ -233,49 +247,89 @@ impl Directive {
             if parts.next().is_some() {
                 return None;
             }
-            let in_span = if let Some(part) = span_part {
+
+            let (in_span, field_value) = if let Some(part) = span_part {
                 let mut parts = part.split(']');
                 let (part0, part1) = (parts.next(), parts.next());
                 if part1 != Some("") {
                     return None;
                 }
-                part0
+                // part0
+
+                if let Some(part0) = part0 {
```

Comment:
We probably should just bite the bullet at this point and replace the hand-written parser with a regex; this is really getting out of hand.

(i'm sorry for inflicting this code on you)

Reply:
no problem! I have written way worse parsers :P